### PR TITLE
DBZ-4605 Test MySQL partial update

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-4605

This will for for MySQL 8 only after new binlog client with https://github.com/osheroff/mysql-binlog-connector-java/pull/64 is used,

### CODE REVIEW

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlConnectorJsonIT.java [hunk 1]
```diff
@@ -157,6 +158,42 @@ public void shouldConsumeAllEventsFromDatabaseUsingSnapshot() throws SQLExceptio
...
+
+        try (MySqlTestConnection conn = MySqlTestConnection.forTestDatabase(DATABASE.getDatabaseName())) {
+            conn.execute(
+                    "CREATE TABLE IF NOT EXISTS deals ( id int(11) unsigned NOT NULL AUTO_INCREMENT, company_id int(11) unsigned DEFAULT NULL, title varchar(255) DEFAULT NULL, custom_fields json DEFAULT NULL, PRIMARY KEY (id), KEY idx_company_id (company_id)) ENGINE=InnoDB DEFAULT CHARSET=utf8",
+                    "INSERT INTO deals (title, custom_fields) VALUES ('test', '{\"17fc9889474028063990914001f6854f6b8b5784\":\"test_field_for_remove_fields_behaviour_2\",\"1f3a2ea5bc1f60258df20521bee9ac636df69a3a\":{\"currency\":\"USD\"},\"4f4d99a438f334d7dbf83a1816015b361b848b3b\":{\"currency\":\"USD\"},\"9021162291be72f5a8025480f44bf44d5d81d07c\":\"test_field_for_remove_fields_behaviour_3_will_be_removed\",\"9b0ed11532efea688fdf12b28f142b9eb08a80c5\":{\"currency\":\"USD\"},\"e65ad0762c259b05b4866f7249eabecabadbe577\":\"test_field_for_remove_fields_behaviour_1_updated\",\"ff2c07edcaa3e987c23fb5cc4fe860bb52becf00\":{\"currency\":\"USD\"}}')",
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlConnectorJsonIT.java:186]: @jpechane, could you add line breaks to the JSON docs? Will make it a bit easier to read.

**Review by gunnarmorling — APPROVED** [review]
LGTM, only one suggestion inline.

**Comment by jpechane** [comment]
@gunnarmorling Done

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlConnectorJsonIT.java [hunk 2]
```diff
@@ -157,6 +158,57 @@ public void shouldConsumeAllEventsFromDatabaseUsingSnapshot() throws SQLExceptio
...
+        }
+        final SourceRecords records = consumeRecordsByTopic(3);
+        assertThat(records.recordsForTopic(DATABASE.topicForTable("deals")).size()).isEqualTo(2);
+        final SourceRecord update = records.allRecordsInOrder().get(2);
+        assertThat(((Struct) update.value()).getStruct("after").getString("custom_fields")).isEqualTo(
+                "{"
+                        + "\"1f3a2ea5bc1f60258df20521bee9ac636df69a3a\":{\"currency\":\"USD\"},"
+                        + "\"4f4d99a438f334d7dbf83a1816015b361b848b3b\":{\"currency\":\"USD\"},"
+                        + "\"9021162291be72f5a8025480f44bf44d5d81d07c\":\"test_field_for_remove_fields_behaviour_3_will_be_removed\","
+                        + "\"9b0ed11532efea688fdf12b28f142b9eb08a80c5\":{\"currency\":\"USD\"},\"e65ad0762c259b05b4866f7249eabecabadbe577\":\"test_field_for_remove_fields_behaviour_1_updated\","
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlConnectorJsonIT.java:206]:
```suggestion
                        + "\"9b0ed11532efea688fdf12b28f142b9eb08a80c5\":{\"currency\":\"USD\"},"
                        + "\"e65ad0762c259b05b4866f7249eabecabadbe577\":\"test_field_for_remove_fields_behaviour_1_updated\","
```

**Comment by gunnarmorling** [comment]
Thx!

**Pull request merged**